functional/sgx/run.sh

```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -x
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+script_path=$(dirname "$0")
+source "${script_path}/../../lib/common.bash"
+
+tmp_data_dir="$(mktemp -d)"
+rootfs_tar="${tmp_data_dir}/rootfs.tar"
+trap cleanup EXIT
+
+# kata-runtime options
+HYPERVISOR=${HYPERVISOR:-"qemu"}
+
+cleanup() {
+	clean_env_ctr
+	sudo rm -rf "${tmp_data_dir}"
+}
+
+pull_rootfs() {
+	# pull and export busybox image in tar file
+	local image="quay.io/prometheus/busybox:latest"
+	sudo -E ctr i pull ${image}
+	sudo -E ctr i export "${rootfs_tar}" "${image}"
+	sudo chown ${USER}:${USER} "${rootfs_tar}"
+	sync
+}
+
+create_bundle() {
+	local bundle_dir="$1"
+	mkdir -p "${bundle_dir}"
+
+	# extract busybox rootfs
+	local rootfs_dir="${bundle_dir}/rootfs"
+	mkdir -p "${rootfs_dir}"
+	local layers_dir="$(mktemp -d)"
+	tar -C "${layers_dir}" -pxf "${rootfs_tar}"
+	for ((i=0;i<$(cat ${layers_dir}/manifest.json | jq -r ".[].Layers | length");i++)); do
```

Comment:
Trying to decide if this is scary clever or just scary :smile:

Reply:
After seeing several comments on this file which I did not touch, decided to look at the Git Hub diffs. And indeed, the PR marks the file as being changed when my local diff does not:

```
git diff --stat main
 .ci/setup_env_centos.sh |   1 +
 .ci/setup_env_fedora.sh |   1 +
 .ci/setup_env_ubuntu.sh |   1 +
 README.md               |  55 ++++++++
 cmd/kata-local-ci       | 344 ++++++++++++++++++++++++++++++++++++++++++++++++
 5 files changed, 402 insertions(+)
```

Not sure why exactly.